frontend/src/hooks/domain/useSpaceAccess.ts

```diff
@@ -1,29 +1,67 @@
 import { useEffect, useState } from 'react';
 import { authService } from '../../apis/services/auth.service';
+import type { SpaceAccessType } from '../../types/space.type';
 
-const useSpaceAccess = (hostId: number | undefined) => {
-  // TODO : 전역에서 내려주는 hostId로 변경
+interface UseSpaceAccessProps {
+  spaceHostId: number | undefined;
+  spaceType: SpaceAccessType | undefined;
+}
+
+const useSpaceAccess = ({ spaceHostId, spaceType }: UseSpaceAccessProps) => {
   const [hasAccess, setHasAccess] = useState(false);
-  const [isLoadingAccess, setIsLoadingAccess] = useState(true);
+  // TODO : 전역에서 내려주는 hostId로 변경
+  const [hostId, setHostId] = useState<number | undefined>(undefined);
+  type LoadingStateType = 'pending' | 'loading' | 'success' | 'error';
+  const [accessLoadingState, setAccessLoadingState] =
+    useState<LoadingStateType>('pending');
 
+  const checkNoAuthAccess = async () => {
+    if (spaceType === 'PUBLIC') {
+      setHasAccess(true);
+      setAccessLoadingState('success');
+      return;
+    }
+    setHasAccess(false);
+    setAccessLoadingState('success');
+  };
+
+  const checkAuthAndAccess = async (visitorId: number) => {
+    if (spaceType === 'PUBLIC') {
+      setHasAccess(true);
+      setAccessLoadingState('success');
+      setHostId(visitorId);
+      return;
+    }
+    setHasAccess(visitorId === spaceHostId);
+    setHostId(visitorId);
+    setAccessLoadingState('success');
+  };
```

Comment:
이 부분 한 함수로 수정할 수 있지 않을까요?


```suggestion
  const checkAuthAccess = async (visitorId?: number) => {
    const isPublic = spaceType === 'PUBLIC';
    const hasPermission = isPublic || visitorId === spaceHostId;
  
    setHasAccess(hasPermission);
    setAccessLoadingState('success');
    if (visitorId) setHostId(visitorId);
  };
```

Reply:
감사뿡💨 드립니다 👍